src/hip/CUDACore/radixSort.h

```diff
@@ -177,8 +183,11 @@ __device__ __forceinline__ void radixSortImpl(
       __syncthreads();
       if (bin >= 0)
         assert(c[bin] >= 0);
-      if (threadIdx.x == 0)
+      if (threadIdx.x == 0) {
         ibs -= sb;
+        // Workaround for problems in radixSort_t.
+        ibs2 = ibs;
```

Comment:
which problem?

Reply:
where ibs2 is used?

Reply:
I suspect what is needed is a threadfence

Reply:
https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#memory-fence-functions
RocM memory model maybe even weaker than CUDA

Reply:
Using a threadfence instead of the temporary variable works.